Develop2

chriskamumu opened the pull request.

chriskamumu committed:
added gitignore and pom files

chriskamumu committed:
added User, UserCredentials and AuthorizedUser classed in model package

chriskamumu committed:
added AbstractRestAssuredClient and AuthClient in client package

chriskamumu committed:
added tests for register: RegisterTest and RegisterWithoutRequiredFieldTest classes

chriskamumu committed:
added tests for login

chriskamumu committed:
added tests for edit

chriskamumu committed:
added OrdersClient and IngredientsClient,
created tests for creation orders

chriskamumu committed:
added tests for getting user orders;
refactored another tests and cliens: removed class AuthorizedUser and changed this to String field;
corrected parametrized tests in EditByUnauthorizedUserTest: added register by user (to try to edit it without authorization)

chriskamumu committed:
added allure: added dependencies and plugin in pom file, added names of tests and steps. And overrided toString() for model's classes for informative display in a report

chriskamumu committed:
add allure report

pomegranate282922 commented on src/test/java/ru/yandex/burgers/CreateOrderTest.java:
```diff
@@ -0,0 +1,121 @@
+        ordersClient = new OrdersClient();
+        IngredientsClient ingredientsClient = new IngredientsClient();
+        ingredientsList = ingredientsClient.get().extract().path("data._id");
+        accessToken = authClient.register(new User("krkr1219_rand@mail.ru", "pass", "name")).extract().path("accessToken");
```
Так нехорошо. Если вдруг случится сбой и клиент не удалится, тесты будут падать потому что не смогут создать такого юзера. Рандомизируй значения!

pomegranate282922 commented on src/test/java/ru/yandex/burgers/EditByUnauthorizedUserTest.java:
```diff
@@ -0,0 +1,71 @@
+    private final String newName;
+
+    private AuthClient authClient;
+    String accessToken = "";
```
А этот почему не приватный?

pomegranate282922 commented on src/test/java/ru/yandex/burgers/EditByUnauthorizedUserTest.java:
```diff
@@ -0,0 +1,71 @@
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> getTestData(){
+        User user =  new User("random2_kr_email@mail.ru", "pass", "name");
```
Везде рандомизируй!

pomegranate282922 commented on src/test/java/ru/yandex/burgers/EditUserTest.java:
```diff
@@ -0,0 +1,74 @@
+    @DisplayName("check editing of user's login and name by authorized user")
+    public void testEditLoginAndNameByAuthorizedUserReturnsSuccessTrue(){
+        User user = new User("test_kr2@mail.ru", "pass", "name");
+        String accessToken = authClient.register(user).extract().path("accessToken");
```
Можно сразу записать в поле

pomegranate282922 commented on src/test/java/ru/yandex/burgers/LoginTest.java:
```diff
@@ -0,0 +1,76 @@
+    @DisplayName("check login with incorrect password")
+    public void testLoginWithIncorrectPasswordReturnsSuccessFalse(){
+        User user = new User("user_kr2_rand@mail.ru", "pass", "name");
+        accessToken = authClient.register(user).assertThat().statusCode(SC_OK).extract().path("accessToken");
```
Общие начала лучше выносить в before для уменьшения дублирования кода

pomegranate282922 commented on src/test/java/ru/yandex/burgers/RegisterTest.java:
```diff
@@ -0,0 +1,68 @@
+
+        this.accessToken = authClient.register(user).statusCode(SC_OK).extract().path("accessToken");
+
+        ValidatableResponse responseOfRefister = authClient.register(user);
```
А если он-таки создастся ( поймает баг?) - нужно удалить этого юзера тоже!

pomegranate282922 commented on src/test/java/ru/yandex/burgers/RegisterWithoutRequiredFieldTest.java:
```diff
@@ -0,0 +1,57 @@
+                .statusCode(SC_FORBIDDEN)
+                .body("success", equalTo(false))
+                .body("message", equalTo("Email, password and name are required fields"));
+    }
```
Здесь тоже ты везде рассчитываешь на зелёные тесты! А ведь может пойматься баг! Предусмотри удаление и в этом случае!( анализируй код ответа)

pomegranate282922 reviewed (commented):
Все хорошо, но есть необходимость доработать. Рандомизируй тестовые данные, предусмотри удаление тестовых данных в случае пойманного бага.

chriskamumu committed:
added class UserUtils to generate random test data;
added removal of users to tests where they are not expected to be created: RegisterWithoutRequiredFieldTest and RegisterTest (testRegisterUserThatAlreadyExistsReturnsSuccessFalse);
fixed another remarks.

chriskamumu committed:
added new allure report

chriskamumu committed:
add allure report

chriskamumu committed:
added lombok

chriskamumu committed:
add allure report

pomegranate282922 commented on src/main/java/ru/yandex/burgers/client/AuthClient.java:
```diff
@@ -0,0 +1,63 @@
+                .body(user)
+                .when()
+                .log().all()
+                .post(AUTH_PATH + "/register")
```
Вообще хорошо бы все урлы вынести в отдельный класс с константами. Так гораздо удобнее работать с проектом.

pomegranate282922 reviewed (approved):
Зачёт!
